dsn_test.go

```diff
@@ -1639,3 +1640,58 @@ func TestExtractAccountName(t *testing.T) {
 		})
 	}
 }
+
+func TestUrlDecodeIfNeeded(t *testing.T) {
+	testcases := map[string]string{
+		"query_tag":             "query_tag",
+		"%24my_custom_variable": "$my_custom_variable",
+	}
+	for param, expected := range testcases {
+		t.Run(param, func(t *testing.T) {
+			decodedParam := urlDecodeIfNeeded(param)
+			if decodedParam != expected {
+				t.Fatalf("urlDecodeIfNeeded returned unexpected response (%v), should be %v", decodedParam, expected)
+			}
+		})
+	}
+}
+
+func TestUrlDecodeIfNeededE2E(t *testing.T) {
+	customVarName := "CUSTOM_VARIABLE"
+	customVarValue := "test"
+	myQueryTag := "mytag"
+
+	cfg := &Config{
+		Account:  os.Getenv("SNOWFLAKE_TEST_ACCOUNT"),
+		User:     os.Getenv("SNOWFLAKE_TEST_USER"),
+		Password: os.Getenv("SNOWFLAKE_TEST_PASSWORD"),
+		Params:   map[string]*string{"$" + customVarName: &customVarValue, "query_tag": &myQueryTag},
+	}
+	mydsn, err := DSN(cfg)
+	if err != nil {
+		t.Fatalf("TestUrlDecodeIfNeededE2E failed to create DSN from Config: %v, err: %v", cfg, err)
+	}
+	db, err := sql.Open("snowflake", mydsn)
+	if err != nil {
+		t.Fatalf("TestUrlDecodeIfNeededE2E failed to connect. %v, err: %v", mydsn, err)
+	}
+	defer db.Close()
+	query := "SHOW /* gosnowflake TestUrlDecodeIfNeededE2E */ VARIABLES;"
```

Comment:
we don't actually need it, i was thinking about if there's any reason we want to identify where is coming from , using only server-side tools (like query history etc), this would allow for it

but indeed isn't necessary, so removing it